### PR TITLE
activate Maven enforcer Upper bound dependencies check

### PR DESCRIPTION
**Pull request by olamy**
And fix some....

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -153,6 +153,7 @@
                 <versionOsgiRule implementation="org.eclipse.jetty.toolchain.enforcer.rules.RequireOsgiCompatibleVersionRule" />
                 <versionRedhatRule implementation="org.eclipse.jetty.toolchain.enforcer.rules.RequireRedhatCompatibleVersionRule" />
                 <versionDebianRule implementation="org.eclipse.jetty.toolchain.enforcer.rules.RequireDebianCompatibleVersionRule" />
+                <requireUpperBoundDeps/>
```
Comment (joakime) [inline pom.xml]: So this is the heart of the change?  the rest is just resolving issues that it created?

Comment (olamy) [reply pom.xml]: yes it is so simple :) I would say resolving the "issues finally discovered" :P

File: jetty-osgi/test-jetty-osgi/pom.xml [hunk 2]
```diff
@@ -576,12 +588,6 @@
           <version>${project.version}</version>
           <scope>test</scope>
         </dependency>
-        <dependency>
-          <groupId>org.conscrypt</groupId>
-          <artifactId>conscrypt-openjdk-uber</artifactId>
```
Comment (joakime) [inline jetty-osgi/test-jetty-osgi/pom.xml]: Where did this go?  Is it not used?

File: pom.xml [hunk 3]
```diff
@@ -894,7 +895,7 @@
       <dependency>
         <groupId>javax.annotation</groupId>
         <artifactId>javax.annotation-api</artifactId>
-        <version>1.2</version>
+        <version>1.3</version>
```
Comment (joakime) [inline pom.xml]: @janbartel should we stay with javax.annotation at 1.2 for jetty 9.4.x?

File: jetty-osgi/test-jetty-osgi/pom.xml (hunk 2 above)
Comment (olamy) [reply jetty-osgi/test-jetty-osgi/pom.xml]: well actually this dependency (and few others) is declared twice:
- in main dependencies section
- in the java9 profile dependencies section
so as it's really needed we only have to declare it once.
(just figure out we can cleanup more here but will do an other pr)

File: pom.xml (hunk 3 above)
Comment (olamy) [reply pom.xml]: the idea was to fix ``` [INFO] --- maven-enforcer-plugin:3.0.0-M2:enforce (enforce-java) @ jetty-http-spi ---
[WARNING] Rule 6: org.apache.maven.plugins.enforcer.RequireUpperBoundDeps failed with message:
Failed while enforcing RequireUpperBoundDeps. The error(s) are [
Require upper bound dependencies error for javax.annotation:javax.annotation-api:1.2 paths to dependency are:
+-org.eclipse.jetty:jetty-http-spi:9.4.14-SNAPSHOT
  +-com.sun.xml.ws:jaxws-rt:2.3.0.2
    +-javax.annotation:javax.annotation-api:1.2 (managed) <-- javax.annotation:javax.annotation-api:1.3 ```
it's not really important as it's used for testing purpose.

**Pull request merged**